some messages are still hardcoded in english

CS still has some output events that are hardcoded in the code and are only in English. Some already have translations in message files, but we don't use them.
We translate check messages to their specific language so non-english users can understand the issue, we should do the same with every output of CS (minus XML attributes and such) so there is no confusion.

Examples:
https://github.com/checkstyle/checkstyle/blob/e18195de775c9ee707f31760820bc45cd1c9aa91/src/main/java/com/puppycrawl/tools/checkstyle/DefaultLogger.java#L130-L143
https://github.com/checkstyle/checkstyle/blob/e18195de775c9ee707f31760820bc45cd1c9aa91/src/main/java/com/puppycrawl/tools/checkstyle/Main.java#L192

@rnveach @romani 

I tried running this command -> `mvn clean integration-test failsafe:verify -DargLine='-Duser.language=de -Duser.country=DE'` but DefaultLoggerTest didn't fail as it was supposed to because the message is hardcoded in the test:
```
assertTrue(output.contains("Error auditing myfile"));
```
But when the same option -> `-Duser.language=de -Duser.country=DE` is set in the POM file the test failed:
```
[INFO] Running com.puppycrawl.tools.checkstyle.DefaultLoggerTest
[ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.001 s <<< FAILURE! - in com.puppycrawl.tools.checkstyle.DefaultLoggerTest
[ERROR] testCtor(com.puppycrawl.tools.checkstyle.DefaultLoggerTest)  Time elapsed: 0 s  <<< FAILURE!
java.lang.AssertionError
	at com.puppycrawl.tools.checkstyle.DefaultLoggerTest.testCtor(DefaultLoggerTest.java:43)
```

I don't understand the difference between the two. I was assuming both give the same result, either UTs fails or passes.

--- a/config/suppressions.xml
+++ b/config/suppressions.xml
@@ -92,6 +92,7 @@
   <suppress checks="ClassFanOutComplexity" files="[\\/]Main\.java"/>
   <suppress checks="ClassFanOutComplexity" files="CheckerTest\.java"/>
   <suppress checks="ClassFanOutComplexity" files="Checker\.java"/>
+  <suppress checks="ClassFanOutComplexity" files="AbstractAutomaticBean\.java"/>
   <suppress checks="ClassFanOutComplexity" files="XdocsPagesTest\.java"/>
   <!-- a lot of GUI elements is OK -->
   <suppress checks="ClassDataAbstractionCoupling" files="(TreeTable|MainFrame)\.java"/>

--- a/src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java
@@ -25,7 +25,6 @@ import java.net.URI;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
-import java.util.Locale;
 import java.util.StringTokenizer;
 import java.util.regex.Pattern;
 
@@ -232,8 +231,9 @@ public abstract class AbstractAutomaticBean
                 final PropertyDescriptor descriptor =
                         PropertyUtils.getPropertyDescriptor(this, key);
                 if (descriptor == null) {
-                    final String message = String.format(Locale.ROOT, "Property '%s' "
-                            + "does not exist, please check the documentation", key);
+                    final String message = getLocalizedMessage(
+                        AbstractAutomaticBean.class,
+                        "AbstractAutomaticBean.doesNotExist", key);
                     throw new CheckstyleException(message);
                 }
             }
@@ -246,13 +246,15 @@ public abstract class AbstractAutomaticBean
             // as we do PropertyUtils.getPropertyDescriptor before beanUtils.copyProperty,
             // so we have to join these exceptions with InvocationTargetException
             // to satisfy UTs coverage
-            final String message = String.format(Locale.ROOT,
-                    "Cannot set property '%s' to '%s'", key, value);
+            final String message = getLocalizedMessage(
+                AbstractAutomaticBean.class,
+                "AbstractAutomaticBean.cannotSet", key, value);
             throw new CheckstyleException(message, exc);
         }
         catch (final IllegalArgumentException | ConversionException exc) {
-            final String message = String.format(Locale.ROOT, "illegal value '%s' for property "
-                    + "'%s'", value, key);
+            final String message = getLocalizedMessage(
+                AbstractAutomaticBean.class,
+                "AbstractAutomaticBean.illegalValue", value, key);
             throw new CheckstyleException(message, exc);
         }
     }
@@ -299,10 +301,29 @@ public abstract class AbstractAutomaticBean
     protected void setupChild(Configuration childConf)
             throws CheckstyleException {
         if (childConf != null) {
-            throw new CheckstyleException(childConf.getName() + " is not allowed as a child in "
-                    + configuration.getName() + ". Please review 'Parent Module' section "
-                    + "for this Check in web documentation if Check is standard.");
+            final String message = getLocalizedMessage(
+                AbstractAutomaticBean.class,
+                "AbstractAutomaticBean.disallowedChild", childConf.getName(),
+                configuration.getName());
+            throw new CheckstyleException(message);
         }
+    }
+    /**
+     * Extracts localized messages from properties files.
+     *
+     * @param caller the {@link Class} used to resolve the resource bundle
+     * @param messageKey the key pointing to localized message in respective properties file.
+     * @param args the arguments of message in respective properties file.
+     * @return a string containing extracted localized message
+     */
+
+    private static String getLocalizedMessage(Class<?> caller,
+                                              String messageKey, Object... args) {
+        final LocalizedMessage localizedMessage = new LocalizedMessage(
+            Definitions.CHECKSTYLE_BUNDLE, caller,
+                    messageKey, args);
+
+        return localizedMessage.getMessage();
     }
 
     /** A converter that converts a string to a pattern. */

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/messages.properties
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/messages.properties
@@ -1,3 +1,7 @@
+AbstractAutomaticBean.cannotSet=Cannot set property ''{0}'' to ''{1}''
+AbstractAutomaticBean.disallowedChild={0} is not allowed as a child in {1}. Please review ''Parent Module'' section for this Check in web documentation if Check is standard.
+AbstractAutomaticBean.doesNotExist=Property ''{0}'' does not exist, please check the documentation
+AbstractAutomaticBean.illegalValue=illegal value ''{0}'' for property ''{1}''
 Checker.cacheFilesException=Unable to persist cache file.
 Checker.error=Error was thrown while processing {0}
 Checker.finishLocalSetup=if no custom moduleFactory is set, moduleClassLoader must be specified

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/messages_de.properties
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/messages_de.properties
@@ -1,3 +1,7 @@
+AbstractAutomaticBean.cannotSet=Eigenschaft ''{0}'' kann nicht auf ''{1}'' gesetzt werden
+AbstractAutomaticBean.disallowedChild={0} ist als Kind in {1} nicht zulässig. Bitte überprüfen Sie den Abschnitt ''Übergeordnetes Modul'' für diese Prüfung in der Web-Dokumentation, wenn die Prüfung Standard ist.
+AbstractAutomaticBean.doesNotExist=Eigenschaft ''{0}'' existiert nicht, bitte überprüfen Sie die Dokumentation
+AbstractAutomaticBean.illegalValue=Ungültiger Wert ''{0}'' für Eigenschaft ''{1}''
 Checker.cacheFilesException=Cache-Datei konnte nicht persistiert werden.
 Checker.error=Fehler wurde während der Verarbeitung ausgelöst {0}
 Checker.finishLocalSetup=wenn kein Brauch moduleFactory eingestellt ist, moduleClassLoader \

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/messages_es.properties
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/messages_es.properties
@@ -1,3 +1,7 @@
+AbstractAutomaticBean.cannotSet=No se puede establecer la propiedad ''{0}'' a ''{1}''
+AbstractAutomaticBean.disallowedChild={0} no está permitido como hijo en {1}. Por favor, revise la sección ''Módulo Padre'' para esta comprobación en la documentación web si la comprobación es estándar.
+AbstractAutomaticBean.doesNotExist=La propiedad ''{0}'' no existe, por favor revise la documentación
+AbstractAutomaticBean.illegalValue=Valor ''{0}'' no válido para la propiedad ''{1}''
 Checker.cacheFilesException=No se pudo persistir el archivo de caché.
 Checker.error=Se produjo un error durante el procesamiento {0}
 Checker.finishLocalSetup=si no hay costumbre moduleFactory EstÃ¡ establecido, \

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/messages_fi.properties
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/messages_fi.properties
@@ -1,3 +1,7 @@
+AbstractAutomaticBean.cannotSet=Ominaisuutta ''{0}'' ei voi asettaa arvoon ''{1}''
+AbstractAutomaticBean.disallowedChild={0} ei ole sallittu lapsena kohteessa {1}. Tarkista tämän tarkistuksen ''Isäntämoduuli''-osio verkkodokumentaatiosta, jos tarkistus on vakio.
+AbstractAutomaticBean.doesNotExist=Ominaisuutta ''{0}'' ei ole olemassa, tarkista dokumentaatio
+AbstractAutomaticBean.illegalValue=Virheellinen arvo ''{0}'' ominaisuudelle ''{1}''
 Checker.cacheFilesException=Välimuistitiedostoa ei voitu tallentaa pysyvästi.
 Checker.error=Virhe tapahtui käsittelyn aikana {0}
 Checker.finishLocalSetup=jos ei ole tapaa moduleFactory on asetettu, moduleClassLoader \

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/messages_fr.properties
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/messages_fr.properties
@@ -1,3 +1,7 @@
+AbstractAutomaticBean.cannotSet=Impossible de définir la propriété ''{0}'' à ''{1}''
+AbstractAutomaticBean.disallowedChild={0} n'est pas autorisé en tant qu'enfant dans {1}. Veuillez consulter la section ''Module Parent'' pour cette vérification dans la documentation Web si la vérification est standard.
+AbstractAutomaticBean.doesNotExist=La propriété ''{0}'' n'existe pas, veuillez consulter la documentation
+AbstractAutomaticBean.illegalValue=Valeur ''{0}'' illégale pour la propriété ''{1}''
 Checker.cacheFilesException=Impossible de persister le fichier de cache.
 Checker.error=Une erreur a été levée lors du traitement {0}
 Checker.finishLocalSetup=si pas de coutume moduleFactory est réglé, \

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/messages_ja.properties
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/messages_ja.properties
@@ -1,3 +1,7 @@
+AbstractAutomaticBean.cannotSet=プロパティ ''{0}'' を ''{1}'' に設定できません
+AbstractAutomaticBean.disallowedChild={0} は {1} の子として許可されていません。チェックが標準である場合は、Webドキュメントでこのチェックの「親モジュール」セクションを確認してください
+AbstractAutomaticBean.doesNotExist=プロパティ ''{0}'' は存在しません。ドキュメントを確認してください。
+AbstractAutomaticBean.illegalValue=プロパティ ''{1}'' の値 ''{0}'' は不正です
 Checker.cacheFilesException=キャッシュファイルを永続化できませんでした。
 Checker.error=処理中にエラーが発生しました {0}
 Checker.finishLocalSetup=習慣がなければ moduleFactory 設定されています, \

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/messages_pt.properties
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/messages_pt.properties
@@ -1,3 +1,7 @@
+AbstractAutomaticBean.cannotSet=Não é possível definir a propriedade ''{0}'' para ''{1}''
+AbstractAutomaticBean.disallowedChild={0} não é permitido como filho em {1}. Por favor, revise a seção ''Módulo Pai'' para esta verificação na documentação da web se a verificação for padrão.
+AbstractAutomaticBean.doesNotExist=A propriedade ''{0}'' não existe, por favor, verifique a documentação
+AbstractAutomaticBean.illegalValue=Valor ''{0}'' ilegal para a propriedade ''{1}''
 Checker.cacheFilesException=Não foi possível persistir o arquivo de cache.
 Checker.error=Ocorreu um erro durante o processamento {0}
 Checker.finishLocalSetup=se nÃ£o houver costume moduleFactory estÃ¡ definido, \

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/messages_ru.properties
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/messages_ru.properties
@@ -1,3 +1,7 @@
+AbstractAutomaticBean.cannotSet=Не удается установить свойство ''{0}'' в ''{1}''
+AbstractAutomaticBean.disallowedChild={0} не допускается в качестве дочернего элемента в {1}. Пожалуйста, просмотрите раздел ''Родительский модуль'' для этой проверки в веб-документации, если проверка стандартная.
+AbstractAutomaticBean.doesNotExist=Свойство ''{0}'' не существует, пожалуйста, проверьте документацию
+AbstractAutomaticBean.illegalValue=Недопустимое значение ''{0}'' для свойства ''{1}''
 Checker.cacheFilesException=Не удалось сохранить файл кэша.
 Checker.error=Произошла ошибка во время обработки {0}
 Checker.finishLocalSetup=если нет обычая moduleFactory установлен, \

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/messages_tr.properties
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/messages_tr.properties
@@ -1,3 +1,7 @@
+AbstractAutomaticBean.cannotSet=''"{0}'' \u00f6zelli\u011fi ''{1}'' olarak ayarlanam\u0131yor
+AbstractAutomaticBean.disallowedChild={0}, {1} i\u00e7inde bir alt \u00f6\u011fe olarak izin verilmiyor. L\u00fctfen, Check standart ise web belgelerindeki bu Check i\u00e7in ''Ana Mod\u00fcl'' b\u00f6l\u00fcm\u00fcn\u00fc inceleyin.
+AbstractAutomaticBean.doesNotExist=''{0}'' \u00f6zelli\u011fi mevcut de\u011fil, l\u00fctfen belgeleri kontrol edin
+AbstractAutomaticBean.illegalValue=''{1}'' \u00f6zelli\u011fi i\u00e7in ge\u00e7ersiz de\u011fer ''{0}''
 Checker.cacheFilesException=�nbellek dosyasi kalici hale getirilemedi.
 Checker.error=Isleme sirasinda bir hata olustu {0}
 Checker.finishLocalSetup=özel değilse moduleFactory ayarlandı, moduleClassLoader belirtilmeli

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/messages_zh.properties
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/messages_zh.properties
@@ -1,3 +1,7 @@
+AbstractAutomaticBean.cannotSet=无法将属性 ''{0}'' 设置为 ''{1}''
+AbstractAutomaticBean.disallowedChild={0} 不允许作为 {1} 的子级。如果该检查是标准检查，请在 Web 文档中查看该检查的“父模块”部分。
+AbstractAutomaticBean.doesNotExist=属性 ''{0}'' 不存在，请检查文档
+AbstractAutomaticBean.illegalValue=属性 ''{1}'' 的值 ''{0}'' 非法
 Checker.cacheFilesException=无法持久化缓存文件。
 Checker.error=处理过程中发生了错误 {0}
 Checker.finishLocalSetup=如果没有定制 moduleFactory 已设置, moduleClassLoader 必须指定

--- a/src/test/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBeanTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBeanTest.java
@@ -138,7 +138,7 @@ public class AbstractAutomaticBeanTest {
                     .fail();
         }
         catch (CheckstyleException exc) {
-            final String expected = "Cannot set property 'exceptionalMethod' to '123.0'";
+            final String expected = "Cannot set property 'exceptionalMethod' to '123'";
             assertWithMessage("Invalid exception cause, should be: ReflectiveOperationException")
                     .that(exc)
                     .hasCauseThat()

